gen3analysis/routes/survival.py

```diff
@@ -1,241 +1,259 @@
+        demographic = case.get("demographic", [])
+        if not demographic:
+            continue
 
-    return Datum(days, demographic.get("vital_status", "").lower() == "alive", meta)
+        demo = demographic[0]
+        days_to_death = demo.get("days_to_death")
+        diagnoses = case.get("diagnoses", [{}])[0]
+        days_to_follow_up = diagnoses.get("days_to_last_follow_up")
 
+        # Use days_to_death if available, otherwise use days_to_follow_up
+        duration = days_to_death if days_to_death is not None else days_to_follow_up
 
-def make_data(ds, case):
-    return list(filter(bool, (make_datum(d["node"], case) for d in ds)))
-
-
-def transform(data):
-    r = [
-        # default value is [{}] to ensure that if there is no diagnose but there is days_to_death and vital_status
-        # make_data function still return the not None value
-        make_data(
-            c["node"].get("diagnoses", {}).get("hits", {}).get("edges", {}), c["node"]
-        )
-        for c in data["edges"]
-        if "diagnoses" in c["node"] or "demographic" in c["node"]
-    ]
-    return [item for sublist in r for item in sublist]
-
-
-def prepare_donor(donor, estimate):
-    donor["survivalEstimate"] = estimate
-    donor["id"] = donor["meta"]["id"]
-    donor["submitter_id"] = donor["meta"]["submitter_id"]
-    donor["project_id"] = donor["meta"]["project_id"]
-    donor.pop("meta", None)
-    return donor
-
-
-def prepare_result(result):
-    items = [item.to_json_dict() for item in result]
+        if duration is not None:
+            records.append(
+                {
+                    "duration": duration,
+                    "event": int(
+                        demo.get("vital_status", "").lower() != "alive"
+                    ),  # 1 if dead, 0 if alive
+                    "case_id": case.get("_case_id"),
+                    "submitter_id": case.get("submitter_id"),
+                    "project_id": case.get("project_id"),
+                }
+            )
 
-    return {
-        "meta": {"id": id(result)},
-        "donors": [
-            prepare_donor(donor, interval.get("cumulativeSurvival"))
-            for interval in items
-            for donor in interval["donors"]
-        ],
-    }
+    return pd.DataFrame(records)
 
 
-def get_curve(
-    filters,
-    gdc_graphql_client,
-    req_opts: Optional[Dict] = None,
+async def get_curve(
+    filters, gen3_graphql_client, auth, req_opts: Optional[Dict] = None
 ):
-    filters = [
-        filters,
-        {
-            "op": "or",
-            "content": [
-                {
-                    "op": "and",
-                    "content": [
-                        {
-                            "op": ">",
-                            "content": {
-                                "field": "demographic.days_to_death",
-                                "value": 0,
+    queryFilter = {
+        "and": [
+            filters,
+            {
+                "and": [
```

Comment:
You should be able to simplify this by removing the "and" list that only has 1 item:
```
    queryFilter = {
        "and": [
            filters,
            {
                "or": [
                    {
                        "nested": {
                            ">": {"days_to_death": 0},
                            "path": "demographic",
                        }
                    },
                    {
                        "nested": {
                            ">": {"days_to_last_follow_up": 0},
                            "path": "diagnoses",
                        }
                    },
                ]
            },
        ]
    }
```